test/src/main/java/org/axonframework/test/fixture/AxonTestFixture.java

```diff
@@ -0,0 +1,594 @@
+import org.axonframework.commandhandling.GenericCommandMessage;
+import org.axonframework.configuration.ApplicationConfigurer;
+import org.axonframework.configuration.NewConfiguration;
+import org.axonframework.eventhandling.EventMessage;
+import org.axonframework.eventhandling.EventSink;
+import org.axonframework.eventhandling.GenericEventMessage;
+import org.axonframework.messaging.Message;
+import org.axonframework.messaging.MessageTypeResolver;
+import org.axonframework.messaging.MetaData;
+import org.axonframework.messaging.unitofwork.AsyncUnitOfWork;
+import org.axonframework.messaging.unitofwork.ProcessingContext;
+import org.axonframework.test.aggregate.Reporter;
+import org.axonframework.test.matchers.FieldFilter;
+import org.axonframework.test.matchers.IgnoreField;
+import org.axonframework.test.matchers.MapEntryMatcher;
+import org.axonframework.test.matchers.MatchAllFieldFilter;
+import org.axonframework.test.matchers.Matchers;
+import org.axonframework.test.matchers.PayloadMatcher;
+import org.axonframework.test.saga.CommandValidator;
+import org.hamcrest.CoreMatchers;
+import org.hamcrest.Description;
+import org.hamcrest.Matcher;
+import org.hamcrest.StringDescription;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.concurrent.CompletableFuture;
+import java.util.function.Consumer;
+import java.util.function.Function;
+import java.util.function.UnaryOperator;
+import java.util.stream.Stream;
+
+import static org.axonframework.test.matchers.Matchers.deepEquals;
+import static org.hamcrest.CoreMatchers.*;
+
+/**
+ * Fixture for testing Axon Framework application. The fixture can be configured to use your whole application
+ * configuration or just a portion of that (single module or component). The fixture allows the execution of
+ * given-when-then style.
+ *
+ * @author Allard Buijze
+ * @author Steven van Beelen
+ * @author Mitchell Herrijgers
+ * @author Mateusz Nowak
+ * @since 5.0.0
+ */
+public class AxonTestFixture implements AxonTestPhase.Setup {
+
+    private final NewConfiguration configuration;
+    private final Customization customization;
+    private final MessageTypeResolver messageTypeResolver;
+    private final RecordingCommandBus commandBus;
+    private final RecordingEventSink eventSink;
+
+    private AxonTestFixture(NewConfiguration configuration, UnaryOperator<Customization> customization) {
+        this.customization = customization.apply(new Customization());
+        this.configuration = configuration;
+        this.messageTypeResolver = configuration.getComponent(MessageTypeResolver.class);
+        this.commandBus = (RecordingCommandBus) configuration.getComponent(CommandBus.class);
+        this.eventSink = (RecordingEventSink) configuration.getComponent(EventSink.class);
+    }
+
+    public static AxonTestPhase.Setup with(ApplicationConfigurer<?> configurer) {
```

Comment:
Nit: I would like some non-null checks on these parameters so that we can provide clear messages.